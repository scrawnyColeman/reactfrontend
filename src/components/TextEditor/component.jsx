import React from 'react';
import AceEditor from 'react-ace';

import 'ace-builds/src-noconflict/mode-javascript';
import 'ace-builds/src-noconflict/mode-java';
import 'ace-builds/src-noconflict/theme-monokai';

const style = {
    margin: `auto`,
};
const TextEditor = ({
    value,
    readOnly = true,
    mode = 'javascript',
    height = '20vh',
    maxLines = 15,
    highlightActiveLine = false,
    minLines = 0,
    placeholder = '',
    fontSize = 14,
}) => (
    <AceEditor
        mode={mode}
        theme="monokai"
        name="blah2"
        fontSize={fontSize}
        highlightActiveLine={highlightActiveLine}
        showPrintMargin={true}
        showGutter={true}
        value={value}
        height={height}
        style={style}
        readOnly={readOnly}
        maxLines={maxLines}
        minLines={minLines}
        placeholder={placeholder}
    />
);
export default TextEditor;
