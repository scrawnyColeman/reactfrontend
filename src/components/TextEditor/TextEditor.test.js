import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import TextEditor from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('TextEditor component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<TextEditor value={'Value'} />);
        expect(wrapper).toMatchSnapshot();
    });
});
