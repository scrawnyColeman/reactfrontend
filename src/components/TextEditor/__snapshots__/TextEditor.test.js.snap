// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor component component matches snapshot 1`] = `
<ReactAce
  cursorStart={1}
  editorProps={Object {}}
  enableBasicAutocompletion={false}
  enableLiveAutocompletion={false}
  enableSnippets={false}
  focus={false}
  fontSize={14}
  height="20vh"
  highlightActiveLine={false}
  maxLines={15}
  minLines={0}
  mode="java"
  name="blah2"
  navigateToFileEnd={true}
  onChange={null}
  onLoad={null}
  onPaste={null}
  onScroll={null}
  placeholder=""
  readOnly={true}
  scrollMargin={
    Array [
      0,
      0,
      0,
      0,
    ]
  }
  setOptions={Object {}}
  showGutter={true}
  showPrintMargin={true}
  style={
    Object {
      "margin": "auto",
    }
  }
  tabSize={4}
  theme="monokai"
  value="Value"
  width="500px"
  wrapEnabled={false}
/>
`;
